server/blob/blob-cassandra/src/main/java/org/apache/james/blob/cassandra/cache/CassandraCacheConfiguration.java

```diff
@@ -0,0 +1,91 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.blob.cassandra.cache;
+
+import java.time.Duration;
+import java.util.Optional;
+
+import com.google.common.base.Preconditions;
+
+public class CassandraCacheConfiguration {
+
+    public static class Builder {
+        private static final Duration DEFAULT_READ_TIMEOUT = Duration.ofSeconds(50);
+        private static final Duration DEFAULT_TTL = Duration.ofDays(7);
+        private static final int DEFAULT_BYTE_THRESHOLD_SIZE = 8 * 1024;
+
+        private Optional<Duration> readTimeout = Optional.empty();
+        private Optional<Integer> sizeThresholdInBytes = Optional.empty();
+        private Optional<Duration> ttl = Optional.empty();
+
+        public Builder timeOut(Duration timeout) {
+            Preconditions.checkNotNull(timeout, "'Threshold size' must not to be null");
+            Preconditions.checkArgument(timeout.toMillis() > 0, "'Threshold size' needs to be positive");
```

Comment:
no max value for timeout?

Reply:
I guess, the range is `30 - 60` sec?

Reply:
> I guess, the range is 30 - 60 sec?

Why being so strict? Integer.MAX_VALUE should be a good max value to require imo (as casting might break stuff)

Reply:
We want to retrieve data from the cache as fast as possible right, should we want to wait 2 mins or 5 mins for query it?

Reply:
> We want to retrieve data from the cache as fast as possible right, should we want to wait 2 mins or 5 mins for query it?

True, but maybe someone would have good reasons to configure higher values? Might have a different definition of fast. We should IMO not be too restrictive.

Reply:
I agree, i propose bring it to configuration properties.

Reply:
We don't even know how long MAX_VALUE lasts.
Put whatever value you want but as a readable Duration please

Reply:
I'm ok with 1 Hour.

Reply:
ok, same on me.